test/helpers/wrappers.go

```diff
@@ -104,6 +104,25 @@ func CurlWithHTTPCode(endpoint string, optionalValues ...interface{}) string {
 		CurlConnectTimeout, endpoint)
 }
 
+// CurlWithRetries returns the string representation of the curl command that
+// retry the request if transient problems occur. Parameter retries are indicated the maximum retry times.
+// If the endpoint already contain "curl", the function will add --retry flag at the end of the command and return.
+// If the endpoint didn't contain "curl", the function will generate the command with --retry flag and return.
+func CurlWithRetries(endpoint string, retries int, optionalValues ...interface{}) string {
```

Comment:
I think it makes a lot of sense to refactor this, but I feel that this functions shows too many implementation details and nesting `CurlFail` into `CurlWithRetries` is not necessary.

I would like `CurlWithRetries` to accept the same arguments as `CurlFail` with addition of `--retry` flag.

I think this function should call `CurlFail` and then perform string replacement on it that would replace `curl` with `curl --retry %d`.

Reply:
Thanks for reviewing. I think the function will become something like this:
``CurlWithRetries`` can use ``failed`` flag to distinguish CurlFail and non-CurlFail.
```
func CurlWithRetries(endpoint string, retries int, failed bool, optionalValues ...interface{}) string {
	if failed {
		return fmt.Sprintf(
			`%s --retry %d`,
			CurlFail(endpoint, optionalValues...), retries)
	}
	if len(optionalValues) > 0 {
		endpoint = fmt.Sprintf(endpoint, optionalValues...)
	}
	return fmt.Sprintf(
		`curl --path-as-is -s  -D /dev/stderr --output /dev/stderr --retry %d %s`,
		retries, endpoint)
}
```

Reply:
Looks good. One nit on argument names - `failed` arg should be called `fail`, as this is equivalent of curl `--fail` flag.

Also please make sure that adding `--retry %d` at the end of command always works as intended (although we will probably figure this out when running CI).

Reply:
Updated! Please take a look.